pkg/controller/resourceclaim/controller.go

```diff
@@ -991,3 +975,59 @@ func claimPodOwnerIndexFunc(obj interface{}) ([]string, error) {
 	}
 	return keys, nil
 }
+func getAdminAccessMetricLabel(claim *resourceapi.ResourceClaim) string {
+	if claim == nil {
+		return "false"
+	}
+	for _, request := range claim.Spec.Devices.Requests {
+		if ptr.Deref(request.AdminAccess, false) {
+			return "true"
+		}
+	}
+	return "false"
+}
+
+func newCustomCollector(rcLister resourcelisters.ResourceClaimLister, adminAccessFunc func(*resourceapi.ResourceClaim) string) metrics.StableCollector {
+	return &customCollector{
+		rcLister:        rcLister,
+		adminAccessFunc: adminAccessFunc,
+	}
+}
+
+type customCollector struct {
+	metrics.BaseStableCollector
+	rcLister        resourcelisters.ResourceClaimLister
+	adminAccessFunc func(*resourceapi.ResourceClaim) string
+}
+
+var _ metrics.StableCollector = &customCollector{}
+
+func (collector *customCollector) DescribeWithStability(ch chan<- *metrics.Desc) {
+	ch <- resourceclaimmetrics.NumResourceClaimsDesc
+}
+
+func (collector *customCollector) CollectWithStability(ch chan<- metrics.Metric) {
+	allocateMetrics := make(map[string]map[string]int)
+	rcList, err := collector.rcLister.List(labels.Everything())
+	if err != nil {
+		klog.FromContext(context.Background()).Error(err, "failed to list resource claims for metrics collection")
```

Comment:
Pass a logger into `newCustomCollector` and use that logger here.

In tests you can use
```
tCtx := ktesting.Init(t)
newCustomCollector(tCtx.Logger(), ...)
```

Reply:
done. PTAL